app/controllers/projects/blob_controller.rb

```diff
@@ -146,10 +146,19 @@ def editor_variables
 
     @file_path =
       if action_name.to_s == 'create'
+        if params[:file_upload].present?
+          params[:file_name] = params[:file_upload].original_filename
+        end
         File.join(@path, File.basename(params[:file_name]))
       else
         @path
       end
+     
+    if params[:file_upload].present?
+      params[:content] = params[:file_upload].read
+      params[:encoding] = 'text'
```

Comment:
@stanhu thanks for your reply. I already got dropzone working with the client side, and I am able to upload the file to repository now. One thing left is that I need to combine the "dropzone file" with the "commit message" in a single form, and I am trying to solve this problem now. It will be good if you have any suggestions, thank you!